device/cuda/src/clusterization/measurement_sorting_algorithm.cu

```diff
@@ -32,9 +32,11 @@ measurement_sorting_algorithm::operator()(
         m_copy.get().get_size(measurements_view);
 
     // Sort the measurements in place
-    thrust::sort(thrust::cuda::par.on(stream), measurements_view.ptr(),
-                 measurements_view.ptr() + n_measurements,
-                 measurement_sort_comp());
+    thrust::sort(
+        thrust::cuda::par(std::pmr::polymorphic_allocator(&(m_mr.main)))
```

Comment:
Strange that you needed to be so very explicit. :thinking: `std::pmr::polymoprhic_allocator` doesn't put `explicit` on its constructor for sure. That's how we can create `vecmem::vector` objects without always writing out the full allocator name in their constructors.

I guess I'll try `thrust::cuda::par(&(m_mr.main)).on(stream)` tomorrow, to see if that would also work...